### PR TITLE
Jacoco plugin integration

### PR DESCRIPTION
**Pull request by tharindu-bandara**
## Purpose
Generate unit test coverage information

## Goals
Integrate Jacoco Maven plugin with Maven Surefire plugin to generate unit test coverage information

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1633,6 +1633,63 @@
...
+                    <version>0.8.0</version>
+                    <executions>
+                        <execution>
+                            <id>default-prepare-agent-by-coverage-enforcer</id>
+                            <goals>
+                                <goal>prepare-agent</goal>
+                            </goals>
+                            <configuration>
+                                <propertyName>argLine</propertyName>
+                                <destFile>${project.build.directory}/jacoco.exec</destFile>
```
Comment (nuwanw) [inline pom.xml]: add the target directory in to the path

File: pom.xml [hunk 2]
```diff
@@ -1633,6 +1633,63 @@
...
+                            </configuration>
+                        </execution>
+                        <execution>
+                            <id>default-report-by-coverage-enforcer</id>
+                            <goals>
+                                <goal>report</goal>
+                            </goals>
+                            <configuration>
+                                <dataFile>${project.build.directory}/jacoco.exec</dataFile>
```
Comment (nuwanw) [inline pom.xml]: add the target directory into the path

File: pom.xml (hunk 1 above)
Comment (tharindu-bandara) [reply pom.xml]: Default target folder is used as the target directory

File: pom.xml (hunk 2 above)
Comment (tharindu-bandara) [reply pom.xml]: Default target folder is used as the target directory

**Pull request merged**